lib/node_modules/@stdlib/math/base/special/hyp2f1/lib/main.js

```diff
@@ -0,0 +1,285 @@
+	* AMS55 #15.3.3
+	*/
+	if ( d <= -1.0 && !( !isIntD && s < 0.0 ) && !( negIntA || negIntB ) ) {
+		return pow( s, d ) * hyp2f1( c - a, c - b, c, x );
+	}
+
+	// The series diverges
+	if ( d <= 0.0 && x === 1.0 && !( negIntA || negIntB ) ) {
+		return PINF;
+	}
+
+	if ( ax < 1.0 || x === -1.0 ) {
+		if ( b === c ) {
+			// 2F1(a,b;b;x) = (1-x)**(-a)
+			return pow( s, -a );
+		}
+		if ( a === c ) {
+			// 2F1(a,b;a;x) = (1-x)**(-b)
+			return pow( s, -b );
+		}
+	}
+
+	if ( negIntC ) {
+		// Check if termination before explosion
+		if ( negIntA && ( ia > ic ) ) {
+			y = hyt2f1( a, b, c, x, err );
+			return y.value;
+		}
+		if ( negIntB && ( ib > ic ) ) {
+			y = hyt2f1( a, b, c, x, err );
+			return y.value;
+		}
+		return PINF;
+	}
+
+	// Function is a polynomial, so try the power series expansion
+	if ( negIntA || negIntB )	{
+		y = hyt2f1( a, b, c, x, err );
+		return y.value;
+	}
+
+	/**
+	* This transform has a pole for integer b - a and may cause large cancellation errors near |1/x| = 1.
+	* AMS55 #15.3.7
+	*/
+	if ( x < -2.0 && !isInteger( t1 ) ) {
+		p = hyp2f1( a, 1.0 - c + a, 1.0 - b + a, 1.0 / x );
+		q = hyp2f1( b, 1.0 - c + b, 1.0 - a + b, 1.0 / x );
+		p *= pow( -x, -a );
+		q *= pow( -x, -b );
+		t1 = gamma( c );
+		s = t1 * gamma( b - a ) / ( gamma( b ) * gamma( c - a ) );
+		y = t1 * gamma( a - b ) / ( gamma( a ) * gamma( c - b ) );
+		return ( s * p ) + ( y * q );
+	}
+	if ( x < -1.0 ) {
+		// Transformation based on AMS55 #15.3.4
+		if ( abs( a ) < abs( b ) ) {
+			return pow( s, -a ) * hyp2f1( a, c - b, c, x / ( x - 1.0 ) );
+		}
+		// Transformation based on AMS55 #15.3.5
+		return pow( s, -b ) * hyp2f1( b, c - a, c, x / ( x - 1.0 ) );
+	}
+
+	// The series diverges for |x| > 1 if above checks fail
+	if ( ax > 1.0 ) {
+		return PINF;
+	}
+
+	p = c - a;
+	r = c - b;
+	negIntCaOrCb = isNonPositiveInteger( p ) || isNonPositiveInteger( r );
+
+	// If |x| == 1.0
+	if ( ax === 1.0 ) {
+		if ( x > 0.0 ) {
+			if ( negIntCaOrCb ) {
+				if ( d >= 0.0 ) {
+					// Transformation based on AMS55 #15.3.3
+					y = hys2f1( c - a, c - b, c, x, err );
+					return pow( s, d ) * y.value;
+				}
+				return PINF;
+			}
+			if ( d <= 0.0 ) {
+				return PINF;
+			}
+			// Transformation based on AMS55 #15.3.6
+			return gamma( c ) * gamma( d ) / ( gamma( p ) * gamma( r ) );
+		}
+		if ( d <= -1.0 ) {
+			return PINF;
+		}
+	}
+
+	/* Conditionally make d > 0 by recurrence on c
+	* AMS55 #15.2.27
+	*/
+	if ( d < 0.0 ) {
+		// Try the power series first:
+		y = hyt2f1( a, b, c, x, err );
+		if ( y.error < ETHRESH ) {
+			return y.value;
+		}
+		// If the power series fails, then apply the recurrence
+		y = y.value;
+		err = 0.0;
+		aid = 2.0 - id;
+		e = c + aid;
+		d2 = hyp2f1( a, b, e, x );
+		d1 = hyp2f1( a, b, e + 1.0, x );
+		q = a + b + 1.0;
+		for ( i = 0; i < aid; i++ ) {
+			r = e - 1.0;
+
+			// eslint-disable-next-line max-len
+			y = (((e * (r - (((2.0 * e) - q) * x))) * d2) + (((e - a) * (e - b) * x) * d1)) / (e * r * s);
```

Comment:
```suggestion
			y = (((e * (r - (((2.0 * e) - q) * x))) * d2) + (((e - a) * (e - b) * x) * d1)) / (e * r * s);
```

General comment: how do we (I) think about whitespace?

- Use negative space to drawn attention to what matters and to group related symbols.

E.g.,

```
x[ i%len ]
```

Why no spaces around `%`? Because, in general, it is useful to indicate that `i%len` is a single "unit". Namely, the operation which results in an index.

Why spaces in `x[ ]`? Because `x[i%len]` is harder to immediately grok where a variable ends and another begins. In `x[i%len]`, there are two pieces of salient info: 1) the array as conveyed by `x[` and 2) the indexing operation. Negative space can help delineate those two things.

So, for the above equation, what are the salient bits of info? Right now, you use whitespace to group parentheses and highlight operators. Maybe that is the right choice, but you could also use whitespace to group related operations/terms. E.g., `( e*r*s )`.